db/upgrade.php

```diff
@@ -72,6 +72,18 @@ function xmldb_auth_saml2_upgrade($oldversion) {
         upgrade_plugin_savepoint(true, 2016031701, 'auth', 'saml2');
     }
 
+    if ($oldversion < 2016071201) {
+        // Update plugin configuration settings from auth_saml2 to auth/saml2
+        $rs = $DB->get_recordset_select('config_plugins', 'plugin = ?', array('auth_saml2'));
+        foreach ($rs as $record) {
+            $record->plugin = 'auth/saml2';
+            $DB->update_record('config_plugins', $record);
```

Comment:
Maybe I'm being a bit paranoid, but I think we should $DB->delete_record() on the old data, and then set_config($key, $config->$key, 'auth/saml2'); on the new